zerodep/src/main/java/com/lightbend/microprofile/reactive/streams/zerodep/Signal.java

```diff
@@ -5,24 +5,20 @@
 
 /**
  * A signal, used by inlets and outlets.
- *
- * A signal represents a callback to an inlet or outlet listener. All callbacks should be wrapped in a signal to
- * allow signals to be enqueued, so that they can be executed in an unrolled fashion, rather than recursively.
- *
- * Implementations of signal are only allocated when a stream or sub stream starts. They should carry no state, any
- * state associated with the signal should be held by the inlet or outlet. This ensures that no allocations are done
- * of signals while running the stream.
+ * <p>
+ * A signal is used to wrap callbacks to inlet and outlet listeners. Rather than invoking these directly, they are
+ * enqueued to be invoked in an unrolled fashion, after the current callback is finished. This ensures that stages do
+ * not have to handle reentrant invocations, when the stage invokes something on a inlet our outlet, it can be sure
+ * that its state before invocation hasn't changed after the invocation returns. This also solves blowing the stack
```

Comment:
"solves blowing the stack" is understandable but a bit weird wording for javadoc documentation, reword?